inc/spbc-auth.php

```diff
@@ -486,6 +489,50 @@ function spbc_2fa__key_remove_old()
     $spbc->save('data');
 }
 
+/**
+ * Handle for check if rate limit is passed
+ *
+ * @return bool
+ */
+function spbc_2fa_is_rate_limit_pass()
+{
+    $rateLimit = get_option('spbc_rate_limit_2fa', [
```

Comment:
Why not use $spbc->data['spbc_rate_limit_2fa']? State object could reduce DB calls and I see the state object is initialized at this point.

Reply:
Option spbc_rate_limit_2fa get only once, it is not necessary to share this option in state.